convert Person to ES6

christalee opened the pull request:
This commit converts Person to use ES6 classes instead of React.createClass, a pre-requisite for migrating from React 15 to 16. It uses `constructor`, a JS builtin, to create the initial state - I'm not 100% sure that's the right way to go and would appreciate feedback on that. Otherwise the function body is nearly unchanged, except for a simplification of Checkbox state in `render`.

jasonaowen commented on src/components/Person.js:
```diff
@@ -5,150 +5,136 @@ import store from 'store2';
+
+  render() {
+    let anonymousRender = (
+            <Checkbox checked={this.state.anonValue === true}
```
Big +1 to this! We could go further, even, by inlining the variable - we don't need `anonymousRender` or `noReadRender` any more, and could just use the `<Checkbox>`es directly in the returned JSX.

jasonaowen commented on src/components/Person.js:
```diff
@@ -5,150 +5,136 @@ import store from 'store2';
+class Person extends React.Component {
+
+  constructor(props) {
+    super(props)
```
The references to `this.props` in the constructor can be converted into references to the `props` argument. The call to `super(props)` should set `this.props`, though; just noting this as a step the [`class` codemod takes](https://github.com/reactjs/react-codemod#explanation-of-the-new-es2015-class-transform-with-property-initializers).

jasonaowen commented on src/components/Person.js:
```diff
@@ -5,150 +5,136 @@ import store from 'store2';
-            }
+class Person extends React.Component {
+
+  constructor(props) {
```
This method is kind of a mess. I think you've handled it correctly, but trying to understand it, it appears that one of its props is a complete list of niceties the current user has written? That shouldn't be the data it gets, and it shouldn't have to try to pick out the maybe-present single relevant nicety.

I'd certainly buy the argument that changing that is out of scope for this change, but if you want to solve it this feels like a good time to do so.

christalee replied:
Do you want to move the nicety-selecting logic into PeopleRow, the parent component? I think it is out of scope but I can start thinking about how I might do that.

jasonaowen replied:
I suppose that is what I'm suggesting! And yeah, we can and should handle that separately; good call.

jasonaowen reviewed (changes requested):
This is looking good! I haven't had a chance to test this yet, and I have a few small suggestions below, but the two changes I'll request are to 1. squash the delete-comment commit into the commit that comments the code, and 2. to avoid reindenting as part of a commit with non-whitespace changes.

christalee commented:
I squashed the commits (including the changes) but I'm unimpressed with the fragmented diff. Is this because of whitespace issues? I didn't think I introduced any changes but maybe I did.

jasonaowen reviewed (approved):
> Is this because of whitespace issues? I didn't think I introduced any changes but maybe I did.

That seems right to me - I think your editor reformatted from 4 space indents to 2 space indents, which is why the diff is larger than expected. I'm happy to get our code all up to the same standard, but I've found it to be more effective to make a separate, whitespace-only commit - it's easier for me as an author to think about, and it's easier for me as a reviewer to understand. I've also found that such reformattings are best to do earlier, so that the whitespace commits overwrite as few meaningful changes in `git blame` as possible.

I'd appreciate it if you could split that out to a separate commit, then go ahead and merge.

Thanks again @christalee - great work!

christalee commented:
I made some whitespace-only commits, and the diff is much more readable now! Sorry I didn't get it together to figure that out before I asked for your review. I'll go ahead and merge.
